### DISCUSSION TITLE
About copyright

### DISCUSSION THREAD

**Discussion by Makkkkus** [MinicraftPlus/minicraft-plus-revived#689] [General]
Given the projects roots as a entry to a Ludum Dare contest by Markus "Notch" Persson in 2011, the project is in a precarious sitaution when it comes to copyright. Notch himself has given us permission to continue the project, but he is unsure if the intellectual property was sold to Microsoft in the Mojang acquisition deal (https://x.com/notch/status/1670068644728938498).

In a discussion about a pull request adding copyright headers to each file, Ben did a thorough search of the source code to find original code, as the excerpt shows.

> And I have compared the current code with the original code ([reference source](https://github.com/skeeto/Minicraft)), and summarized as below:
> | Original Source Method/Class | Current Minicraft+ Method/Class | Remarks |
> | -------------------------------- | ----------------------- | -------------------------- |
> | `Game#renderFocusNagger()` | `Renderer#renderFocusNagger()` | Mostly more or less unchanged, but more comments added | `LevelTransitionDisplay` | `LevelTransitionDisplay` | Mostly the same, but the class is small |
> | `LevelGen` | `LevelGen` | Except some lines of additions and changes due to systems, mostly the same as the original |
> | Overrides of `Tile#interact()` | Overrides of `Tile#interact()` | Mostly the same in logic |
> | `InfiniteFallTile` | `InfiniteFallTile` | Mostly the same, but the class is small |
> | `Level#renderBackground()` | `Level#renderBackground()` | Mostly the same, but with changes to values and the size is not big |
> | `ItemEntity#tick()` | `ItemEntity#tick()` | Mostly the same, but more comments added |
> | `Mob#findStartPos()` | `MobAi#checkStartPos()` | Logically similar, but with some different code |
> | `TextParticle#tick()` | `TextParticle#tick()` | Mostly the same, but more comments added |
> 
> Additional remarks:
> - Some menus contain mainly just the transformations of implementation from the original source because of the entry system.
> - Methods contained only or less than 5 simple code lines are not counted in.
> - Sometimes, just more additions of code are added with respect to the original code in some methods, but not counted in.
> 
> The issues about the codebase copyright may not be significant enough.
> 
> In addition, other than the codebase, another concern is about the assets (which may not be licensed under any media licenses), including sprites and audio, may have been copyrighted, but still under the similar concern of licensing as like the original codebase.

_Originally posted by @BenCheung0422 in https://github.com/MinicraftPlus/minicraft-plus-revived/issues/684#issuecomment-2194444309_

Game assets are a little bit more problematic, since a lot of the original art is kept. There have been significant discussion about doing a full overhaul of the in-game textures, but these proved to be unpopular. Although, given the gravity of the situation, changing the texture is something that _should_ be done. This can also be done in a way that doesn't diverge a lot from the original, as these examples below show.

![Image of a sprite-sheet showing items in the game.](https://github.com/MinicraftPlus/minicraft-plus-revived/assets/37084190/346d6639-c5c8-4834-9922-bfa3650fb2f9)
@brickerman. on Discord (https://discord.com/channels/280723930942013440/419258986865360896/1144298159766515713)

![An image of a base located in a valley between two mountains.](https://github.com/MinicraftPlus/minicraft-plus-revived/assets/37084190/8fe19bdc-025d-4c16-8f40-cd47055d7df1)
@randomguy1390 on Discord (https://discord.com/channels/280723930942013440/419258986865360896/1056604284772417587).

I propose it should be a general goal to move away from all original assets and code, while still retaining the style of the original.

**Comment by BenCheung0422** [comment]
Not long ago, I have discussed with @PseudoDistant and @El-Virus about this topic on Discord. The topic was about hosting a service that similar to Minecraft Realms, and eventually talking about the copyright issues with the game. While we are using the copyrighted materials without explicit or clear permissions by someone, either Notch or Microsoft, the project is still under risk. However, the point also mentioned that if Microsoft does not sue on this due to its unpopularity, there might not be issues, but risk instead.

Another perspective is that, this may break the extent of this project being "open source". People may concern about the copyright issues when doing forks or direct modding, but other than that, there might not be many impacts by this on the Minicraft economy.

I did have reworked some of the textures during the time, but the extent might not be enough and my arts may not be really well on this. It is actually possible that if we could completely make our own assets to replace the original assets and put the original assets *the other side* (distribute something like "classic assets" as another resource pack, privately, or similar). The previous discussion of overhaul on textures actually includes the texture style, which some people do not agree on. Therefore, there could either be a consent to have a complete change on assets, or we just leave them there as is.

**Comment by BenCheung0422** [comment]
Hopefully we could have more updates on the assets in 2.3 and eventually all replaced with our own ones to remove the concerns after 2.3. And it is a question to whether provide classic assets as a pack together. By the way, does our GPL license covers media?

**Comment by BenCheung0422** [comment]
I think the media assets can be licensed with either GPL or CC license for convenience. However, the original assets cannot be said so, since they were not licensed by any written document.
